bootstrap/golang/prep-new-repo.sh

```diff
@@ -0,0 +1,112 @@
+cp -r "${input_dirpath}/${LAMBDA_IMPL_DIRNAME}" "${output_dirpath}/"
+cp -r "${input_dirpath}/${SCRIPTS_DIRNAME}" "${output_dirpath}/"
+
+
+# =============================================================================
+#                         Post-Copy Modifications
+# =============================================================================
+# Allow setting the module name programmatically, for testing in CI
+new_module_name="${GO_NEW_MODULE_NAME:-}"
+echo "Go uses Github as a its dependency management store. You'll now need to:"
+echo "  1) Create a new Github repo to contain your Kurtosis Lambda, if you don't have one already"
+echo "  2) Enter the URL of the repo on Github WITHOUT the leading 'https://' below (e.g. 'github.com/my-org/my-repo')"
+echo "NOTE: This value is technically the Go module name. If you're unfamiliar with what this is, you can read more here: https://golang.org/ref/mod"
+echo ""
+while [ -z "${new_module_name}" ]; do
+    read -p "Module name: " new_module_name
+done
+
+# Validation, to save us in case someone changes stuff in the future
+go_mod_filepath="${output_dirpath}/${GO_MOD_FILENAME}"
+if [ "$(grep "^${GO_MOD_MODULE_KEYWORD}" "${go_mod_filepath}" | wc -l)" -ne 1 ]; then
+    echo "Validation failed: Could not find exactly one line in ${GO_MOD_FILENAME} with keyword '${GO_MOD_MODULE_KEYWORD}' for use when replacing with the user's module name" >&2
+    exit 1
+fi
+
+# Replace module names in code (we need the "-i '' " argument because Mac sed requires it)
+existing_module_name="$(grep "module" "${go_mod_filepath}" | awk '{print $2}')"
+if ! sed -i"${SED_INPLACE_FILE_SUFFIX}" "s,${existing_module_name},${new_module_name},g" ${go_mod_filepath}; then
+    echo "Error: Could not replace Go module name in mod file '${go_mod_filepath}'" >&2
+    exit 1
+fi
+# We search for old_module_name/kurtosis-lambda because we don't want the old_module_name/lib entries to get renamed
+if ! sed -i"${SED_INPLACE_FILE_SUFFIX}" "s,${existing_module_name}/${LAMBDA_IMPL_DIRNAME},${new_module_name}/${LAMBDA_IMPL_DIRNAME},g" $(find "${output_dirpath}" -type f); then
+    echo "Error: Could not replace Go module name in code files" >&2
+    exit 1
+fi
+
+# Replacing Docker image name in build script
+# Validation, to save us in case someone changes stuff in the future
+build_script_filepath="${output_dirpath}/${SCRIPTS_DIRNAME}/${BUILD_SCRIPT_FILENAME}"
+if [ "$(grep "^${BUILD_SCRIPT_KEYWORD}" "${build_script_filepath}" | wc -l)" -ne 1 ]; then
+    echo "Validation failed: Could not find exactly one line in ${BUILD_SCRIPT_FILENAME} with keyword '${BUILD_SCRIPT_KEYWORD}' for use when replacing with the user's Docker image name" >&2
+    exit 1
+fi
+
+# Replace Docker image names in code (we need the "-i '' " argument because Mac sed requires it)
+new_image_name="${BUILD_SCRIPT_KEYWORD}=\"${lambda_image}\""
+existing_image_name="$(grep "${BUILD_SCRIPT_KEYWORD}" "${build_script_filepath}" | head -1)"
```

Comment:
ooo this is actually a pretty loose grep - e.g. any line with `IMAGE_NAME` in it will match. I'd prefer a tighter grep of `^IMAGE_NAME=".*"$`.

Though actually, we can just get rid of the `grep` entirely and do `sed -i"${SED_INPLACE_FILE_SUFFIX}" "s,^${BUILD_SCRIPT_KEYWORD}=".*",${BUILD_SCRIPT_KEYWORD}=\"${lambda_image}\" .....etc......`

Reply:
good improvement